File: app/src/main/java/com/amyllykoski/asyncseq/api/MyService.java
package com.amyllykoski.asyncseq.api;

import com.amyllykoski.asyncseq.model.Item;
import com.amyllykoski.asyncseq.model.RestCallback;

import java.util.List;

public interface MyService {
  void getItems(RestCallback<List<Item>> items);
}

File: app/src/main/java/com/amyllykoski/asyncseq/api/MyService.kt
package com.amyllykoski.asyncseq.api

import arrow.core.Either
import com.amyllykoski.asyncseq.model.Item

interface MyService {
  fun getItems(): Either<MyError, List<Item>>
  fun close()
}

sealed class MyError {
  object ServerError : MyError()
  data class AnotherError(val msg: String) : MyError()
}
